crates/uv-resolver/src/version_map.rs

```diff
@@ -25,6 +28,14 @@ pub struct VersionMap {
 
 impl VersionMap {
     /// Initialize a [`VersionMap`] from the given metadata.
+    ///
+    /// Note it is possible for files to have a different yank status per PEP 592 but in the official
+    /// PyPI warehouse this cannot happen.
+    ///
+    /// Here, we track if each file is yanked separately. If a release is partially yanked, the
+    /// unyanked distributions _can_ be used.
```

Comment:
I hope no index implements this, this would be quite confusing to users

Reply:
Fun fact, the spec is about marking a particular file as yanked: https://peps.python.org/pep-0592/.

PyPI has implemented it as only being able to yank an entire release, but that it a PyPI implementation detail.

This has come up a few times on discussing particular behavior pip could adopt around yanking.

Reply:
I see why it needs to be like that on the API level (there are no releases in the API, just files), but i'd strongly prefer it if indexes made this an all or nothing operation per release.

> While this PEP implements yanking at the file level, that is largely due to the shape the simple repository API takes, not a specific decision made by this PEP.

Reply:
Agreed, there has been some recent discussion about requiring behavior on indexes, not just the API, and the concern was being able to find and get buy in from non PyPI indexes (Azure, Gitlab, Artifactory, etc.). Perhaps a good case to put forth as a trial to see if it's possible.